### PR TITLE
fix(receipt): describe real scanner transforms in the evidence-provenance profile

### PR DESCRIPTION
**Pull request by luckyPipewrench**
## What changed

The evidence-provenance transform profile now has typed operations for the transforms the scanner actually performs, plus a gate that fails when a production transform has no typed operation.

Sixteen operations are added, each mapped to the scanner site it describes: `query_unescape`, `invisible_space`, `matching_normalize`, the three liberal decoders, `encoded_token_normalize`, `text_segment`, `html_entity_decode`, `whitespace_compact`, `url_noise_strip`, `ordered_query_concat`, `query_subsequence`, `hostname_dot_remove`, `encoded_run`, and `canary_canonicalize`. A `raw_query` URL component is appended as well.

Existing operations keep their positions. Enum bytes are derived from position, so reordering would silently rewrite every historical commitment preimage; operations 1 through 11 remain byte-for-byte identical and only bytes 12 through 27 are new. The profile digest changes accordingly.

## Why this was needed

The language could not describe what the scanner does, and the gap was not cosmetic. `percent_decode` was specified as `PathUnescape` with at most four passes while the scanner uses `QueryUnescape` with up to five hundred, and handles `+` differently. `base64_decode` accepted only the standard alphabet while the scanner also accepts URL-safe. `invisible_strip` removes invisible characters while several scanner paths replace them with spaces, which produces different bytes. Nothing described `ForMatching`, HTML entity decoding, whitespace compaction, ordered query concatenation, query subsequences, hostname dot removal, or delimiter-split token normalization.

The sharp edge is that the scanner accepts some decodings the profile rejected as non-canonical. Routing the scanner through the shared implementation would have eliminated detections and changed what gets blocked. That is why this lands as a correction to the language rather than a change to the scanner.

## The completeness gate

Adding operations alone would leave the same failure available next time. The profile was originally validated against its own corpus, written from the same design document, so nothing ever compared the vocabulary against production behavior.

`internal/scanner/transform_profile_completeness_test.go` parses non-test scanner and normalization source with Go's AST. It derives requirements from directives attached directly to production transform declarations, confirms those declarations still have production references, and checks every declared operation against the typed vocabulary. It detects a typed operation removed from a live transform, a stale declaration whose production consumer is gone, and a directive naming an operation the vocabulary does not support.

It cannot recognize a brand-new inline or unannotated transform, reflective or interface-dispatched behavior, external-language transforms, or prove runtime branch reachability. Those still need reviewer attention, and the limits are documented in the test itself rather than left implied.

## Scanner behavior is unchanged

No scanner test was modified. The only new test file under `internal/scanner` is the completeness gate. Scanner source changes are behavior-preserving extractions, for example `removeHostnameDots(value)` whose body is exactly the `strings.ReplaceAll` call it replaces, so the gate has named declarations to annotate. The scanner and normalization suites pass unmodified under the race detector.

## Downstream

The digest change invalidates the sixty-three provenance fixtures on the open four-verifier pull request. Those regenerate against the corrected profile after this merges, which is the intended order.

## Summary by CodeRabbit

* **New Features**
  * Added evidence-provenance transformation and proof verification across Python, Rust, and TypeScript.
  * Added provenance CLI commands with structured reports and incomplete-result handling.
  * Added URL, query, encoding, normalization, invisible-character, HTML entity, token, and text transformations.
  * Added raw-query extraction, expanded recipe parameters, bounded processing, and pinned Unicode 15.0.0 behavior.
* **Bug Fixes**
  * Improved hostname normalization while preserving existing behavior.
  * Added fail-closed handling for invalid, incomplete, or unsupported proof data.
* **Documentation**
  * Documented digest pinning and profile lifecycle rules.
* **Tests**
  * Added cross-language conformance and verification coverage.

### CODE REVIEW

File: sdk/verifiers/ts/src/provenance.ts [hunk 1]
```diff
@@ -0,0 +1,646 @@
...
+      ),
+    )
+    .join("");
+}
+function removeUnicode15NonspacingMarks(value: string): string {
+  return [...value].filter((char) => !isUnicode15NonspacingMark(char)).join("");
+}
+function compactUnicode15Whitespace(value: string): string {
+  return [...value].filter((char) => !isUnicode15Whitespace(char)).join("");
+}
```
[the remark this answers is not in the document]
Comment (luckyPipewrench) [reply sdk/verifiers/ts/src/provenance.ts:99]: Rejected, with the measurement.

The premise is that White_Space=Yes holds for U+001C..U+001F, so this verifier would strip characters the Go reference keeps. Measured against the pinned dependency this verifier actually uses, the property is false for all four:

    U+001C White_Space: false
    U+001D White_Space: false
    U+001E White_Space: false
    U+001F White_Space: false
    U+0020 space:       true

So isUnicode15Whitespace already returns false for the four information separators and they are already retained. No divergence exists.

Verified the Go side independently rather than reasoning about it. Feeding the four separators plus an ASCII space through the shared recipe implementation, Go returns codepoints [97 98 99 28 29 30 31 100 101 102]: the separators survive and only the space is compacted. TypeScript produces the same view.

I did implement the suggested exclusion first. Neutralizing it changed no test outcome, which is what exposed the guard as dead code, and the change was reverted. A retention test is added to pin the agreement so this cannot regress silently, and the function now carries a comment recording the measurement so the same suggestion is not re-raised.

File: internal/scanner/transform_profile_completeness_test.go [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+			parsedFiles = append(parsedFiles, parsed)
+		}
+		info := &types.Info{
+			Defs: make(map[*ast.Ident]types.Object),
+			Uses: make(map[*ast.Ident]types.Object),
+		}
+		config := types.Config{Importer: importer.ForCompiler(set, "source", nil)}
+		if _, err := config.Check(directory.packagePath, set, parsedFiles, info); err != nil {
+			t.Fatalf("type-check %s: %v", directory.packagePath, err)
+		}
```
[the remark this answers is not in the document]
Comment (luckyPipewrench) [reply internal/scanner/transform_profile_completeness_test.go:123]: I reproduced this against the current head. importer.Default() makes go test ./internal/scanner -run TestScannerTransformProfileCompleteness -count=1 fail because the standard importer only searches GOROOT/GOPATH and cannot import this module’s internal/config. This test needs the source importer for module-local type checking, so I’m keeping it.

File: sdk/verifiers/rust/src/provenance.rs [hunk 3]
```diff
@@ -0,0 +1,973 @@
...
+                        .and_then(Value::as_str)
+                        .ok_or_else(|| "query component: missing selector".to_string())?;
+                    if selector.is_empty() {
+                        return Err("query component: missing selector".to_string());
+                    }
+                } else if self.fields.contains_key("selector")
+                    || self.fields.contains_key("occurrence")
+                {
+                    return Err(format!("unsupported selector for {}", self.kind));
+                }
```
[the remark this answers is not in the document]
Comment (luckyPipewrench) [reply sdk/verifiers/rust/src/provenance.rs]: I reproduced both branches on the current code. A missing query selector is rejected as query component: missing selector, while a present selector with an unavailable occurrence is rejected as query component: occurrence N unavailable. The focused url_component_surfaces_are_distinct Rust test passes and exercises both messages, so I am keeping the distinction.

[the remark this answers is not in the document]
Comment (luckyPipewrench) [reply sdk/verifiers/rust/src/provenance.rs]: You’re right; my earlier rejection tested the wrong branch. Query selectors were covered, but a non-query component carrying occurrence still fell through to the wrong error. Fixed in b041466b with exact regression coverage for path plus occurrence. Thanks for pushing back.

File: sdk/verifiers/python/src/pipelock_aarp_verify/provenance_proof.py [hunk 4]
```diff
@@ -0,0 +1,702 @@
...
+        try:
+            _validate_intervals(view, [(match[1], match[2]) for match in matches])
+        except ProvenanceError:
+            invalid = _invalid("location", result)
+            if unavailable_source:
+                invalid["view_reproduction"] = "not_checked"
+                invalid["location"] = "not_checked"
+                invalid["match_commitment"] = "not_checked"
+            return invalid
+        result["location"] = "exact_coordinates"
```
[the remark this answers is not in the document]
Comment (luckyPipewrench) [reply sdk/verifiers/python/src/pipelock_aarp_verify/provenance_proof.py:641]: Rejected after reproduction on b041466b. The committed p38 fixture emits the same byte-identical report in Go, Rust, TypeScript, and Python: failure_stage is location and location is not_checked. The cited Rust branch is intermediate per-source state; final missing-source aggregation downgrades the three source-dependent stages. The frozen p38 known answer and four-language comparison agree, so changing Python here would create the divergence described by the finding.

File: cmd/pipelock-verifier/provenance_test.go [hunk 5]
```diff
@@ -0,0 +1,300 @@
...
+				source.SourceOrdinal = uint64(index + 1)
+				source.SourceID = fmt.Sprintf("source-%d", index)
+				signed.Proof.Sources[index] = source
+			}
+		})
+		for len(fixture.Entries)*provenanceMaxSources <= provenanceMaxSourceReferences {
+			appendProvenanceTestEntry(t, &fixture)
+		}
+		assertProvenanceProofStructureRejection(t, fixture)
+	})
```
[the remark this answers is not in the document]
Comment (luckyPipewrench) [reply cmd/pipelock-verifier/provenance_test.go:151]: Fixed in eeb0febf. The test now proves exactly 128 references is accepted and 129 is rejected; neutralizing the guard makes the over-boundary case fail.

File: sdk/verifiers/python/src/pipelock_aarp_verify/provenance.py [hunk 6]
```diff
@@ -0,0 +1,725 @@
...
+        remaining = MAX_CUMULATIVE_PROCESSED_BYTES
+
+        def charge(processed: str) -> None:
+            nonlocal remaining
+            size = len(processed.encode())
+            if charge_fixture_work is not None:
+                charge_fixture_work(size)
+            if size > remaining:
+                raise ProvenanceError("recipe: exceeds cumulative processing budget")
+            remaining -= size
```
[the remark this answers is not in the document]
Comment (luckyPipewrench) [reply sdk/verifiers/python/src/pipelock_aarp_verify/provenance.py:496]: Fixed in eeb0febf, with the class aligned across all four verifiers. The per-recipe profile budget is charged before the fixture-wide budget. Go also needed separate per-recipe and fixture-wide error signals so the two failure stages cannot collapse.

File: sdk/verifiers/rust/src/provenance_proof.rs [hunk 7]
```diff
@@ -0,0 +1,1747 @@
...
+fn parse_artifact(verification: &Map<String, Value>, key: &str) -> ArtifactInput {
+    match optional_string(verification, key) {
+        Ok(None) => ArtifactInput::Absent,
+        Ok(Some(value)) => match decode_b64(value, key) {
+            Ok(bytes) if bytes.len() <= MAX_ARTIFACT_BYTES => ArtifactInput::Bytes(bytes),
+            Ok(_) | Err(_) => ArtifactInput::Invalid,
+        },
+        Err(_) => ArtifactInput::Invalid,
+    }
+}
```
[the remark this answers is not in the document]
Comment (luckyPipewrench) [reply sdk/verifiers/rust/src/provenance_proof.rs]: Fixed in eeb0febf. Non-string artifact fields now fail at proof_structure; invalid or oversized strings retain artifact-stage handling when attested. Neutralizing the type-error propagation made the regression test fail.

**Comment by luckyPipewrench** [comment]
Rejecting the outside-diff suggestion for internal/normalize/provenance.go:645-656. scannerQueryUnescape replays the production scanner's IterativeDecode behavior: a malformed later pass stops decoding and returns the last successfully decoded value. The digest-pinned profile states that behavior explicitly, and all four verifier implementations follow it.

The existing regression pins %25zz to %zz; the focused Go race test passes. Returning an error here would make provenance replay disagree with the scanner and the pinned profile, so no code change is warranted.

**Review by Celest-Blue — APPROVED** [review]
